packages/connections/src/stores/connections-store.ts

```diff
@@ -175,6 +177,10 @@ export function useConnections(
       connectedConnectionInfo.current &&
       connectedDataService.current
     ) {
+      // Update lastUsed date as now and save the connection.
+      connectedConnectionInfo.current.lastUsed = new Date();
+      void connectionStorage.save(connectedConnectionInfo.current);
+
       // After connecting and the UI is updated we notify the rest of Compass.
       void onConnected(
```

Comment:
This function we'll eventually have synchronous since it's just emitting the `data-service-connected` event in compass-home. It's currently async because we convert it to a legacy connection model for two of the plugins to use, which we'll be getting rid of. If it's cool with you I'd like to keep changes around that not in this pr.
If we keep the legacy conversion yes we should definitely catch it as well.

Reply:
ok! and do I understand it right that we won't have to handle a failure in that case? Would it be helpful info to add a line for a possible issue in the logs?

Reply:
Good call, added error handling to it we can remove when we remove that code, as well as a log when the error occurs.